lti_consumer/lti_1p3/extensions/rest_framework/permissions.py

```diff
@@ -0,0 +1,44 @@
+"""
+Django REST Framework extensions for LTI 1.3 & LTI Advantage implementation.
+
+Implements a custom authorization classes to be used by any of the
+LTI Advantage extensions.
+"""
+from rest_framework import permissions
+
+
+class LtiAgsPermissions(permissions.BasePermission):
+    """
+    LTI AGS Permissions.
+
+    This checks if the token included in the request
+    has the allowed scopes to read/write LTI AGS items
+    (LineItems, Results, Score).
+
+    LineItem scopes: https://www.imsglobal.org/spec/lti-ags/v2p0#scope-and-allowed-http-methods
+    Results: Not implemented yet.
+    Score: Not implemented yet.
+    """
+    def has_permission(self, request, view):
+        # Retrieves token from request, which was already checked by
+        # the Authentication class, so we assume it's a sane value.
+        auth_token = request.headers.get('Authorization', '').split()[1]
```

Comment:
@giovannicimolin if ``Authorization`` is empty then ``split()[1]`` will throw exception. If we are certain that ``Authorization`` will not be empty then probably ``request.headers['Authorization'].split()[1]`` is better, because default value has no use here.

Reply:
Improved, thanks!